streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -393,6 +414,66 @@ class ZStreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestR
     } yield released must_=== true
   }
 
+  private def broadcastValues =
+    unsafeRun {
+      Stream.range(0, 5).broadcast(2, 12).use {
+        case s1 :: s2 :: Nil =>
+          for {
+            out1     <- s1.runCollect
+            out2     <- s2.runCollect
+            expected = List(0, 1, 2, 3, 4, 5)
+          } yield (out1 must_=== expected) && (out2 must_=== expected)
+        case _ =>
+          ZIO.fail("Wrong number of streams produced")
+      }
+    }
+
+  private def broadcastErrors =
+    unsafeRun {
+      (Stream.range(0, 1) ++ Stream.fail("Boom")).broadcast(2, 12).use {
+        case s1 :: s2 :: Nil =>
+          for {
+            out1     <- s1.runCollect.either
+            out2     <- s2.runCollect.either
+            expected = Left("Boom")
+          } yield (out1 must_=== expected) && (out2 must_=== expected)
+        case _ =>
+          ZIO.fail("Wrong number of streams produced")
+      }
+    }
+
+  private def broadcastBackPressure =
+    flaky(
+      Stream
+        .range(0, 5)
+        .broadcast(2, 2)
+        .use {
+          case s1 :: s2 :: Nil =>
+            for {
+              ref      <- Ref.make[List[Int]](Nil)
+              _        <- s1.timeout(100.milliseconds).foreach(i => ref.update(i :: _)).ignore
```

Comment:
Hmm. I can't think of anything here. Maybe you can give an example of how you would test that it correctly backpressure here. Otherwise we can just throw it out I think

Reply:
Something like this:
- The source is `Stream.range(0, 5).tap(i => ref.update(i :: _)).broadcast...`
- In the `use`:
```scala
for {
  latch1 <- Promise.make[Nothing, Unit]
  latch2 <- Promise.make[Nothing, Unit]
  fib <- s1.tap(i => latch1.succeed(()) *> latch2.await.when(i == 2)).runDrain.fork
  _ <- latch1.await
  snapshot1 <- ref.get
  _ <- latch2.succeed(())
  _ <- fib.await
  snapshot2 <- ref.get
} yield (snapshot1 must_=== <elems that are not backpressured>) and (snapshot2 must_=== <all elems>)
```
  
```

Reply:
Yeah that works. Thank you!